piptools/writer.py

```diff
@@ -258,15 +277,14 @@ def _format_requirement(
             required_by.add(_comes_from_as_string(ireq))
 
         if required_by:
-            sorted_required_by = sorted(required_by)
-            if len(sorted_required_by) == 1:
-                source = sorted_required_by[0]
-                annotation = "    # via " + source
-            else:
-                annotation_lines = ["    # via"]
-                for source in sorted_required_by:
-                    annotation_lines.append("    #   " + source)
-                annotation = "\n".join(annotation_lines)
-            line = f"{line}\n{comment(annotation)}"
+            annotation = {
+                "split": annotation_style_split,
+                "line": annotation_style_line,
+            }[self.annotation_style](required_by)
+            sep = {"split": "\n    ", "line": "\n    " if ireq_hashes else "  "}[
+                self.annotation_style
+            ]
```

Comment:
IMO this would be much more readable if written as if blocks.
With only two options there is really no need to use dispatch tables.

Reply:
I've redone that as blocks.

Before, an unexpected `self.annotation_style` would naturally raise a `KeyError`, but is now unhandled. I wonder about checking against a tuple or enum in either the writer or the compile script.

Reply:
Oh, but of course we are enforcing the correct style names via the click parameter.